server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -591,7 +591,7 @@ public void onResponse(BulkShardResponse bulkShardResponse) {
                     public void onFailure(Exception e) {
                         // create failures for all relevant requests
                         for (BulkItemRequest request : requests) {
-                            final String indexName = concreteIndices.getConcreteIndex(request.index()).getName();
+                            final String indexName = request.index();
```

Comment:
No need to lookup `Index` from `ConcreteIndices` if we just use it as name in an exception message.

Reply:
I suspect that this one line could give some pretty big memory savings on the coordinating node sometimes. Prior to this change we'd retain a reference to the `ConcreteIndices` (and therefore the `ClusterState`) while the request was ongoing. Now we don't. Great stuff. 🎉